### ISSUE TITLE
Compiler error in Xcode 14.3 in CTLocalDataStore.m file

### ISSUE THREAD

**Issue by tushar127** [CleverTap/clevertap-ios-sdk#240]
Hi, I have installed Xcode 14.3 version in my system & after that when I am trying to build code I am getting compiler error in CTLocalDataStore.m file on the line number 416(method name: syncWithRemoteData).

Error: Non-void function does not return a value in all control paths

I attached screenshot as well. Please check & update the code

![clevertap](https://user-images.githubusercontent.com/107125164/228207982-03026213-1de3-40cf-8281-cf4ade1fc873.jpg)

**Comment by akashvercetti** [comment]
@tushar127 could you please provide us with the clevertap SDK version?

**Comment by tushar127** [comment]
Right now I am using 4.0.1 but I updated it to 4.2.1 but same issue is coming with new version as well

**Comment by akashvercetti** [comment]
We are currently investigating this issue. In the meantime could you please try compiling it on Xcode 14.3 RC2? The 14.3 beta version has some compile issues with a lot of SDKs according to various forums.
The latest 14.3 RC2 version was released a week ago: https://download.developer.apple.com/Developer_Tools/Xcode_14.3_Release_Candidate/Xcode_14.3_Release_Candidate.xip

**Comment by cowboyeric** [comment]
@akashvercetti
This issues just simply related that the function is defined will return a NSDictionary.
But the logic just check if bool "somethingGotUpdated" true, then return nil. If "somethingGotUpdated" still false, the function have no return anything.

TLDR:
Add else return nil. Then fix it.
![Screenshot 2023-03-30 at 12 16 24 PM](https://user-images.githubusercontent.com/20036582/228727992-e4355eb5-f852-4a94-b609-59326deb42ce.png)

**Comment by NiliStein** [comment]
I see there's already a PR that solves this:
https://github.com/CleverTap/clevertap-ios-sdk/pull/242

Waiting for the update as well.
In the meantime, personally we downgraded Xcode back to 14.2 but it's a very temporary solution, I guess.

**Comment by akashvercetti** [comment]
We have released a fix for this in our latest release v.4.2.2 https://github.com/CleverTap/clevertap-ios-sdk/releases/tag/4.2.2 👍
